dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESReadOnlyMonitorTest.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    /**
+     * Method to Test: {@link ESReadOnlyMonitor#start(ReindexEntry, String)}
+     * When: If any of LIVE and WORKING current index are read only
+     * Should: sent large message
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSendLargeMessage() throws DotDataException, DotSecurityException, InterruptedException {
+        final ReindexEntry reindexEntry = mock(ReindexEntry.class);
+        final String cause = "cause";
+
+        final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+
+        final Role adminRole = mock(Role.class);
+        when(roleAPIMock.loadCMSAdminRole()).thenReturn(adminRole);
+
+        final User user = mock(User.class);
+        when(user.getUserId()).thenReturn("1");
+
+        when(roleAPIMock.findUsersForRole(adminRole)).thenReturn(list(user));
+
+        try {
+            putReadOnly(indiciesInfo.getWorking(), true);
+            putReadOnly(indiciesInfo.getLive(), false);
+
+            esReadOnlyMonitor.start(reindexEntry, cause);
+
+            Thread.sleep(100);
+
+            checkLargeMessageSent(indiciesInfo, user);
+            assertEquals(false, ESIndexUtil.isReadOnly(indiciesInfo.getWorking(), indiciesInfo.getLive()));
+        } finally {
+            putReadOnly(indiciesInfo.getWorking(), false);
+        }
+    }
+
+    /**
+     * Method to Test: {@link ESReadOnlyMonitor#start(ReindexEntry, String)}
+     * When: If call start again after the first call is finished
+     * Should: should sent the message again
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSendLargeMessageTwice() throws DotDataException, DotSecurityException, InterruptedException {
+        final ReindexEntry reindexEntry = mock(ReindexEntry.class);
+        final String cause = "cause";
+
+        final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+
+        final Role adminRole = mock(Role.class);
+        when(roleAPIMock.loadCMSAdminRole()).thenReturn(adminRole);
+
+        final User user = mock(User.class);
+        when(user.getUserId()).thenReturn("1");
+
+        when(roleAPIMock.findUsersForRole(adminRole)).thenReturn(list(user));
+
+        try {
+            putReadOnly(indiciesInfo.getWorking(), true);
+            putReadOnly(indiciesInfo.getLive(), false);
+
+            esReadOnlyMonitor.start(reindexEntry, cause);
+            checkLargeMessageSent(indiciesInfo, user);
+
+            esReadOnlyMonitor.start(reindexEntry, cause);
+            checkLargeMessageSent(indiciesInfo, user);
+            assertEquals(false, ESIndexUtil.isReadOnly(indiciesInfo.getWorking(), indiciesInfo.getLive()));
+        } finally {
+            putReadOnly(indiciesInfo.getWorking(), false);
+        }
+    }
+
+    private void checkLargeMessageSent(IndiciesInfo indiciesInfo, User user) throws InterruptedException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'indiciesInfo' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642172)